src/caffeine/prif_private_s.f90

```diff
@@ -185,24 +185,45 @@ pure function caf_elem_len(a) result(a_elem_len) bind(C)
        integer(c_size_t), target :: a_elem_len
      end function
 
-     pure function caf_as_int(ptr) bind(C)
-       !! intptr_t caf_as_int(void* ptr);
-       import c_ptr, c_intptr_t
-       type(c_ptr), intent(in), value :: ptr
-       integer(c_intptr_t) :: caf_as_int
-     end function
-
-     pure function caf_as_c_ptr(i) bind(C)
-       !! void* caf_as_c_ptr(intptr_t i);
-       import c_ptr, c_intptr_t
-       integer(c_intptr_t), intent(in), value :: i
-       type(c_ptr) :: caf_as_c_ptr
-     end function
-
   end interface
 
 contains
 
+  pure function caf_as_int(ptr)
```

Comment:
IIUC Caffeine currently follows the convention that `caf_` names are only used for `BIND(C)` procedures, so it's easy to see at the call-site when we are making a cross-language call (which is probably not subject to normal inlining, and thus potentially a more expensive call). Private Caffeine procedures implemented in Fortran have no prefix (e.g. `assert`, `unimplemented`, `optional_value`), because they are already hidden by the module privacy.

This seems like a useful convention to preserve. Assuming you agree, let's rename these three converted procedures appropriately and update the coding convention in <docs/README-maintainers.md>

Reply:
I agree. I think that is a good convention to have. If @everythingfunctional agrees with that convention, I am happy to update the convention in that doc on a separate branch and PR. Or it can become part of this PR, whichever is best.

To make sure I follow the suggested convention, it would go something like this right?:

_Procedures with the prefix `prif_` are public procedures that are part of the PRIF specification.
Procedures with the prefix `caf_` are internal Caffeine procedures that are `BIND(C)` and whose implementation is in C.
Procedures with no prefix are internal private Caffeine procedures whose implementation is in Fortran._

Reply:
That convention seems reasonable, so I've done as @bonachea suggested.